### PR TITLE
Support Gradle

### LINKED ISSUE #11990: Support Gradle
It's a part of https://github.com/playframework/play-meta/issues/8

### PR DESCRIPTION
**Pull request by ihostage**
## Fixes

Fixes #11990

### CODE REVIEW

**Comment by ihostage** [comment]
So, I implemented _"Aggregating reverse routers"_ feature and finished adoption to Gradle all our Java examples (except gRPC) in https://github.com/playframework/play-samples/issues/467.
I think we are ready to make announcement of beta testing.

**Comment by mkurz** [comment]
Great! Let me take a look tomorrow.

**Comment by aalubin** [comment]
Any update on this?

**Comment by ihostage** [comment]
Hi, @aalubin! 👋 
Yep, today we announced a [start of beta-testing](https://github.com/orgs/playframework/discussions/12338) 😉

**Comment by mkurz** [comment]
@ihostage Please take a look at the commits I pushed:
I renamed the `isChanged` supplier to `triggerReload`, because I think this is a bit clearer and I think it's confusing to have the `changed` variable and an `isChanged` supplier. (I kept `isChanged` in the gradle plugin code however, feel free to rename it also if you think it makes sense there).
Because of #12270 we now also have to impliment the generateJsReverseRouter setting in gradle, I did that, please check the last commit.

File: dev-mode/gradle-plugin/src/main/java/play/gradle/plugin/PlayRoutesPlugin.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                        .getAggregateReverseRoutes()
+                        .get()
+                        .contains(project));
+  }
+
+  private void configureDefaultSettings(final Project project) {
+    var routes = playExtension(project).getRoutes();
+    routes.getNamespaceReverseRouter().convention(false);
+    routes.getGenerateReverseRouter().convention(project.provider(() -> !isAggregated(project)));
+    routes.getGenerateJsReverseRouter().convention(project.provider(() -> !isAggregated(project)));
```
Comment (mkurz) [inline dev-mode/gradle-plugin/src/main/java/play/gradle/plugin/PlayRoutesPlugin.java:82]: @ihostage is this ok for `getGenerateJsReverseRouter? like to also use `project.provider(() -> !isAggregated(project))` as convention?

Comment (ihostage) [reply dev-mode/gradle-plugin/src/main/java/play/gradle/plugin/PlayRoutesPlugin.java:82]: Currently, I think yes. By default, like in Sbt plugin, we should generate a reverse router for non-aggregated projects.

**Pull request merged**